org.jboss.tools.maven.apt.core/src/org/jboss/tools/maven/apt/internal/AbstractAptConfiguratorDelegate.java

```diff
@@ -185,7 +196,7 @@ public void configureProject(IProgressMonitor monitor) throws CoreException {
       }
     }
 
-    Map<String, String> currentOptions = AptConfig.getProcessorOptions(javaProject);
+    Map<String, String> currentOptions = AptConfig.getRawProcessorOptions(javaProject);
```

Comment:
why use getRawProcessorOptions? it seems more limiting than getProcessorOptions

Reply:
getProcessorOptions includes computed options  -classpath, -sourcepath, -s, -d, so the following if() will alway fail. 
also, the computed options are now different for main and test compilation